pkg/datapath/iptables/iptables.go

```diff
@@ -494,17 +494,21 @@ func (m *IptablesManager) SupportsOriginalSourceAddr() bool {
 
 // removeRulesAndIpsets removes iptables rules and ipsets installed by Cilium.
 func (m *IptablesManager) removeRulesAndIpsets(prefix string, quiet bool) {
-	// Set of tables that have had iptables rules in any Cilium version
-	tables := []string{"nat", "mangle", "raw", "filter"}
-	for _, t := range tables {
-		m.removeCiliumRules(t, ip4tables, prefix+"CILIUM_")
+	if option.Config.EnableIPv4 {
+		// Set of tables that have had iptables rules in any Cilium version
+		tables := []string{"nat", "mangle", "raw", "filter"}
+		for _, t := range tables {
+			m.removeCiliumRules(t, ip4tables, prefix+"CILIUM_")
+		}
 	}
 
-	// Set of tables that have had ip6tables rules in any Cilium version
-	if m.haveIp6tables {
-		tables6 := []string{"nat", "mangle", "raw", "filter"}
-		for _, t := range tables6 {
-			m.removeCiliumRules(t, ip6tables, prefix+"CILIUM_")
+	if option.Config.EnableIPv6 {
+		// Set of tables that have had ip6tables rules in any Cilium version
+		if m.haveIp6tables {
+			tables6 := []string{"nat", "mangle", "raw", "filter"}
+			for _, t := range tables6 {
+				m.removeCiliumRules(t, ip6tables, prefix+"CILIUM_")
+			}
```

Comment:
@joamaki Doesn't this mean that if you enable IPv6 in the agent, then disable IPv6 in the agent, then we'll leave a bunch of IPv6 configuration behind? I think it was the way it was before deliberately to try to clean up after Cilium when the user explicitly disables configuration.

Reply:
I think that question was meant for me.

You're probably correct. Looking at how `haveIp6tables` is defined, it seems we will only execute this code if the IPv6 modules are loaded. So something else is causing the error reported in https://github.com/cilium/cilium/issues/18513. I'll revert and ask for more information.

Reply:
Revert sent at https://github.com/cilium/cilium/pull/18768.